nodeadm/internal/feature-gates/feature-gates.go

```diff
@@ -36,14 +33,14 @@ func DefaultFalse(featureGate FeatureGate, featureGates map[FeatureGate]bool) bo
 
 // Discovers all cluster details using a describe call to the eks endpoint and
 // updates the value of the config's `ClusterDetails` in-place
-func PopulateClusterDetails(cfg *api.NodeConfig) error {
+func PopulateClusterDetails(clusterName string, cfg *api.NodeConfig) error {
```

Comment:
This seems like an odd place for this, we don't want the `featuregates` package to turn into a junk drawer of all the impl details of each feature gate. You also create an AWS Config in a few places, it's probably worth doing that once as close to `main` as possible, and maybe move this logic into an `eks` package with a wrapper around the SDK client or something

Reply:
> we don't want the featuregates package to turn into a junk drawer of all the impl details of each feature gate 

+1

ill work on breaking this up and finding a better home